simulator/util/sharedstore/sharedstore.go

```diff
@@ -0,0 +1,93 @@
+package sharedstore
+
+import (
+	"context"
+
+	"golang.org/x/xerrors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/kube-scheduler-simulator/simulator/util"
+)
+
+type Store interface {
+	AddResultStore(store ResultStore, key string)
+	ResisterResultSavingToInformer(informerFactory informers.SharedInformerFactory, client clientset.Interface)
+}
+
+// ResultStore represents the store which is stores data and shared with simulator and scheduler.
+// Fulfilling this interface will allow the stored results to be saved as data in that Pod
+// when the Pod's schedule is complete.
+type ResultStore interface {
```

Comment:
This interface represents the functionality that the resultstore of extender or an existing plugin should fulfill.
To accommodate this change, a major modification has been made to the logic of the Plugin's resultstore.